File: app/src/main/java/it/cammino/risuscito/viewmodels/SalmiIndexViewModel.java
package it.cammino.risuscito.viewmodels;

import android.app.Application;
import android.arch.lifecycle.LiveData;

import java.util.List;

import it.cammino.risuscito.database.RisuscitoDatabase;
import it.cammino.risuscito.database.SalmoCanto;

public class SalmiIndexViewModel extends GenericIndexViewModel {

  private LiveData<List<SalmoCanto>> mIndexResult;


  public SalmiIndexViewModel(Application application) {
    super(application);
  }

  public LiveData<List<SalmoCanto>> getIndexResult() {
    return mIndexResult;
  }

  public void createDb() {
    mDb = RisuscitoDatabase.Companion.getInstance(getApplication());
    // Receive changes
    subscribeToDbChanges();
  }

  private void subscribeToDbChanges() {
    mIndexResult = mDb.salmiDao().getLiveAll();
  }
}

File: app/src/main/java/it/cammino/risuscito/viewmodels/SalmiIndexViewModel.kt
package it.cammino.risuscito.viewmodels

import android.app.Application
import android.arch.lifecycle.LiveData

import it.cammino.risuscito.database.RisuscitoDatabase
import it.cammino.risuscito.database.SalmoCanto

class SalmiIndexViewModel(application: Application) : GenericIndexViewModel(application) {

    var indexResult: LiveData<List<SalmoCanto>>? = null
        private set

    fun createDb() {
        mDb = RisuscitoDatabase.getInstance(getApplication())
        // Receive changes
        subscribeToDbChanges()
    }

    private fun subscribeToDbChanges() {
        indexResult = mDb!!.salmiDao().liveAll
    }
}
